File: python/pythagorean-triplet/2/pythagorean_triplet.py
def triplets_with_sum(N):
    triplets = []
    for big in range(int(N / 2.0) - 1, int((2**0.5 - 1) * N), -1):
        D = (big**2 - N**2 + 2 * N * big)**0.5
        if D == int(D):
            small = int((N - big - D) / 2)
            mid = int((N - big + D) / 2)
            triplets.append([small, mid, big])
    return triplets

File: julia/pythagorean-triplet/1/pythagorean-triplet.jl
function pythagorean_triplets(N)
    triplets = Tuple{Int, Int, Int}[]
    for big in (N ÷ 2 - 1):-1:(round(Int, (√2 - 1) * N))
        D_sq = big^2 - N^2 + 2N * big
        D_sq > 0 || break
        D = √D_sq
        if isinteger(D)
            small = round(Int, ((N - big - D) / 2))
            mid = round(Int, ((N - big + D) / 2))
            push!(triplets, (small, mid, big))
        end
    end
    triplets
end
